enhancements/single-node/single-node-openshift-day2-workers.md

```diff
@@ -0,0 +1,290 @@
+  additional workers recovers after the single control-plane node reboots
+  by running conformance tests post-reboot.
+
+- Add tests on both cloud / UPI that check that a single-node cluster with
+  additional workers recovers after an upgrade by running conformance tests
+  post-upgrade.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+TBD
+
+#### Tech Preview -> GA
+
+TBD
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+In the non-goals section it's mentioned that this enhancement does not apply to
+clusters which have been installed prior to the enhancement, so their upgrade
+is not discussed.
+
+This enhancement, to the best of my knowledge, should have no problems persisting
+across any type cluster upgrades. The Test Plan section describes how this will be
+tested.
+
+### Version Skew Strategy
+
+Does not apply, to the best of my understanding.
+
+### Operational Aspects of API Extensions
+
+This enhancement does not modify/add any API
+
+#### Failure Modes
+
+This enhancement does not modify/add any API
+
+#### Support Procedures
+
+This enhancement does not modify/add any API
+
+## Implementation History
+
+Not yet applicable
+
+## Drawbacks
+
+- This is another change which would make single-node clusters slightly different from
+  multi-node clusters, and any such difference is naturally not ideal.
+
+- The installer maintainers would have to make sure that the default
+  `IngressController` resource generated for single-node bootstrap-in-place
+  installations is similar enough to the `IngressController` generated by the
+  Openshift Cluster Ingress Operator by default in the absence of an existing
+  resource. i.e. our goal here was to simply change the node selector label
+  from "worker" to "master", but to do that we had to create an entire
+  `IngressController`, one that does not get merged with the default one that
+  would've been created by the OpenShift Cluster Ingress Operator otherwise.
+  Any future important additions to the default `IngressController` created by
+  the OpenShift Cluster Ingress Operator would also have to be added to the
+  installer code where it generated the manifest for single-node
+  bootstrap-in-place installations.
+
+- TBD
+
+## Alternatives
+
+- Implement this change in the OpenShift Cluster Ingress Operator rather than the installer.
+  I'm not sure how trivial it is for the OpenShift Cluster Ingress Operator to know that we're
+  doing a single-node bootstrap-in-place installation. It couldn't just inspect the
+  control-plane and infrastructure topology fields, because those have the same value
+  whether you're on AWS (which doesn't suffer from the problems described in this enhancement)
+  or a UPI bootstrap-in-place installation. Maybe it could also look at the "platform"
+  field in the infrastructure CR, then only do this if the platform is "none".
```

Comment:
Beside the fact that SNO on AWS platform isn't supported, we should probably strive to the same Ingress behavior in all platforms.
So if we decide to pin the ingress to the control plane node in none platform we should do the same for other platforms.

Reply:
> Beside the fact that SNO on AWS platform isn't supported

It might be supported in the future so I wouldn't like to dismiss it from the discussion entirely

> we should probably strive to the same Ingress behavior in all platforms.
So if we decide to pin the ingress to the control plane node in none platform we should do the same for other platforms.

Not sure that I agree with this. I'd rather ingress to be as flexible as possible when the circumstances allow it, and only force it to a smaller subset of nodes (the SNO node) when it's necessary. 

I'd argue that it's more important that SNO remains similar to multi-node as much as possible, rather than remain similar to itself across different platforms. And by pinning just when it's necessary you're closer to the former than the latter